valora-inc/wallet
# chore(deps): upgrade firebase dependencies to v19.1.0

### Description

The privacy manifests were included in the firebase iOS [v10.22.0 SDK](https://firebase.google.com/support/release-notes/ios#10.22.0) which is in [v19 of the react native SDK](https://github.com/invertase/react-native-firebase/blob/main/packages/app/CHANGELOG.md#1901-2024-03-07). This is required for Apple privacy policy compliance which goes into effect 1st may.

According to the change log for v18.0.0 and v19.0.0, there should be no breaking changes for us (the breaking changes are around minimum tooling versions).

This PR locks the version to v19.1.0 because this is the last version before the [google `play-services-auth` version is updated to 21](https://github.com/invertase/react-native-firebase/blob/b742500cdd2a2128f4696e4b0b479d901314e2da/packages/app/package.json#L88). [Version 21 of `play-services-auth` removes the Credentials API](https://developers.google.com/android/guides/releases) but unfortunately this is used in `react-native-sms-retriever` which we use in the CPV phone number screen. This lib is not maintained anymore. While this lib declares the version it needs as v17, it seems that 2 versions of `play-services-auth` in one app does not coexist and the newest one is used - the android app crashes on the phone number screen because of the removed Credentials API. Since we cannot upgrade firebase beyond this point until we do something about this sms-retriever lib (probably replace it with an active project, but it's not necessary for this work), I've required the absolute version here.

### Test plan

Rely on CI for app functionality.

Manually test on iOS and Android:
1. remote config values (e.g. the remote config value for `celoNews` should be available in redux in Flipper) ✅
2. rtdb connection (token prices, supercharge remote notification) ✅
3. dynamic link create and decode (generate invite link, verify inviter address on link consume) ✅
4. receive push notification for payment received (using fcm token through firebase messaging dashboard) ✅

### Related issues

- Related to RET-1058

### Backwards compatibility

Y

### Network scalability

Y
